File: app/src/main/java/network/SupabaseClient.java
package network;

import android.util.Log;
import com.example.myapplication.BuildConfig;

import java.io.IOException;
import java.util.HashMap;
import java.util.Map;
import java.util.concurrent.TimeUnit;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class SupabaseClient {
    public static final SupabaseClient INSTANCE = new SupabaseClient();
    public static final String API_KEY = BuildConfig.SUPABASE_KEY;
    private static final String BASE_URL = BuildConfig.BASE_URL;

    private final SupabaseService retrofitService;
    private final FotoService fotoService;

    private SupabaseClient() {
        HttpLoggingInterceptor logging = new HttpLoggingInterceptor();
        logging.setLevel(HttpLoggingInterceptor.Level.BODY);

        OkHttpClient client = new OkHttpClient.Builder()
                .connectTimeout(5, TimeUnit.SECONDS)
                .readTimeout(5, TimeUnit.SECONDS)
                .writeTimeout(5, TimeUnit.SECONDS)
                .addInterceptor(logging)
                .build();

        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl(BASE_URL)
                .client(client)
                .addConverterFactory(GsonConverterFactory.create())
                .build();

        retrofitService = retrofit.create(SupabaseService.class);
        fotoService = retrofit.create(FotoService.class); // 💡 Tambahkan ini
    }

    public SupabaseService getRetrofitService() {
        return retrofitService;
    }

    public FotoService getFotoService() {
        return fotoService;
    }

    public static void updateEmployeePassword(int id, String newPassword) {
        SupabaseService service = INSTANCE.getRetrofitService();

        Map<String, Object> body = new HashMap<>();
        body.put("password", newPassword);

        Call<Void> call = service.updatePassword(
                API_KEY,
                "Bearer " + API_KEY,
                "application/json",
                "eq." + id,
                body
        );

        call.enqueue(new Callback<Void>() {
            @Override
            public void onResponse(Call<Void> call, Response<Void> response) {
                if (response.isSuccessful()) {
                    Log.d("Supabase", "Password updated for id=" + id);
                } else {
                    try {
                        String error = response.errorBody() != null ? response.errorBody().string() : "No error";
                        Log.e("Supabase", "Update failed: " + error);
                    } catch (IOException e) {
                        Log.e("Supabase", "Error reading error body", e);
                    }
                }
            }

            @Override
            public void onFailure(Call<Void> call, Throwable t) {
                Log.e("Supabase", "Network error: " + t.getMessage());
            }
        });
    }
}

File: app/src/main/java/network/SupabaseClient.kt
package network

import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object SupabaseClient {
    const val API_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImlnbHNhYmlscHl2YXJ6ZGFkb2NqIiwicm9sZSI6ImFub24iLCJpYXQiOjE3NDQzODMyMjUsImV4cCI6MjA1OTk1OTIyNX0.jB7ual9xU_xixJW-8I0Ft98RBTl1CZo_cCzlFcVu0Ks"
    private const val BASE_URL = "https://iglsabilpyvarzdadocj.supabase.co/rest/v1/"

    private val logging = HttpLoggingInterceptor().apply {
        level = HttpLoggingInterceptor.Level.BODY
    }

    private val client = OkHttpClient.Builder()
        .addInterceptor(logging)
        .build()

    private val retrofit = Retrofit.Builder()
        .baseUrl(BASE_URL)
        .client(client) // <- tambahkan client yang ada logging-nya
        .addConverterFactory(GsonConverterFactory.create())
        .build()

    val retrofitService: SupabaseService by lazy {
        retrofit.create(SupabaseService::class.java)
    }
}
